explorer/src/utils/index.ts

```diff
@@ -65,26 +65,23 @@ export function countryDataToGridRow(
   return sorted;
 }
 
-export function mixnodeToGridRow(
-  arrayOfMixnodes: MixNodeResponse,
-): MixnodeRowType[] {
+export function mixnodeToGridRow(arrayOfMixnodes: MixNodeResponse): any {
   return !arrayOfMixnodes
     ? []
     : arrayOfMixnodes.map((mn) => {
         const pledge = Number(mn.bond_amount.amount) || 0;
         const delegations = Number(mn.total_delegation.amount) || 0;
         const totalBond = pledge + delegations;
         const selfPercentage = ((pledge * 100) / totalBond).toFixed(2);
-
         return {
           id: mn.owner,
           owner: mn.owner,
-          location: mn?.location?.country_name || '',
           identity_key: mn.mix_node.identity_key || '',
           bond: totalBond || 0,
+          location: mn?.location?.country_name || '',
           self_percentage: selfPercentage,
-          host: mn.mix_node.host || '',
-          layer: mn.layer || '',
+          host: mn?.mix_node?.host || '',
+          layer: mn?.layer || '',
```

Comment:
Just curious as to why some `mn` don't have the optional chaining operator and some do? I would have thought either they'd all need it or none of them would.

Reply:
From memory i think it was that ALL of them have an identity key, owner etc. But there are records that have blank location/null. I will add to my to-do list though to check..